test/functional/test-resources.js

```diff
@@ -1824,6 +1833,67 @@ describe('Resources mutateElement and collapse', () => {
     });
   });
 
+  it('attemptCollapse should not call attemptChangeSize', () => {
```

Comment:
Sorry for going back and forth. But few things I found:
1. `request.callback()` can do something to affect element `layoutBox`. For example we are setting `resource.isFixed` to false when we call `#completeCollapse` in the callback. This won't be an issue for us because we will also manually update element `layoutBox` in `#completeCollapse` call.
2. So I can still go with re-ordering calls, and assert `#completeCollapse` gets called before `#getLayoutBox`.  However just like what I do with `#getScrollHeight` `#getLayoutBox` gets called 2 times in the test path. [here](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L922) and [here](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1023). I fell I will still need to stub `#getLayoutBox` to assert on the second call. which makes no difference between my current test approach.

One other thing that's not related to this PR:
if we resize an fixed element, is it likely that it won't affect other elements layout on the page. Maybe we can add some optimization by not updating `minTop` [here](https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1024) if element is position fixed.